docs/upgrades/v0.14-to-v1-upgrade/examples/src/test/kotlin/examples/ErrorHandlingChanges.kt

```diff
@@ -0,0 +1,66 @@
+package examples
+
+import org.junit.jupiter.api.assertThrows
+import org.partiql.parser.PartiQLParser
+import org.partiql.parser.PartiQLParserException
+import org.partiql.parser.SourceLocation
+import org.partiql.planner.PartiQLPlanner
+import java.util.Random
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ErrorHandlingChanges {
+    /**
+     * The example shows when using {@link PartiQLParser} to parse string statement,
+     * the {@link PartiQLParserException} will be thrown. The parser will stop and throw the exception
+     *  when an error occurs.
+     */
+    @Test
+    fun `Parser error handling`() {
+        val parser = PartiQLParser.default()
+
+        val exception = assertThrows<PartiQLParserException> {
+            try {
+                val parseResult = parser.parse("Create Table 'mytable")
+            } catch (e: PartiQLParserException) {
+                // Rethrow for validating the exception in test
+                throw e
+            }
+        }
+
+        assertTrue { exception.message.contains("extraneous input") }
+        assertEquals("UNRECOGNIZED", exception.tokenType)
```

Comment:
I'd add a comment highlighting that this is a pain point with using the 0.14.9 parser. There's no proper way to extract out error information other than looking at the error message string.